plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2285,7 +2285,7 @@ private void removeOrphanFiles(Table table, ConnectorSession session, SchemaTabl
 
                 validMetadataFileNames.add(fileName(manifest.path()));
                 try (ManifestReader<? extends ContentFile<?>> manifestReader = readerForManifest(table, manifest)) {
-                    for (ContentFile<?> contentFile : manifestReader) {
+                    for (ContentFile<?> contentFile : manifestReader.select(ImmutableList.of("file_path"))) {
```

Comment:
nit: we could extract `ImmutableList.of("file_path")` to a static field - it doesn't need to be recreated every time.

Reply:
JIT can handle micro-optimizations like that